doc/design/katib_on_sqlflow.md

```diff
@@ -70,22 +71,42 @@ Consider the following example program.
 
 ```sql
 SELECT * FROM a, b WHERE a.id = b.id INTO c;
-SELECT * FROM c TO TRAIN model_def WITH objective=multi:softmax, eta=1 LABEL class INTO my_xgb_model;
+SELECT * FROM c TO TRAIN model_def 
+    WITH objective=multi:softmax, eta=1, validation_dataset="select * from d;" 
+    INTO my_xgb_model;
 ```
 
 The `codegen_couler.go` might generate the following Couler program.
 
 ```python
-couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;
-                         SELECT * FROM c INTO temp""")
-couler.maxcompute.export(table="temp", file="/hdfs/temp")
-couler.katib.train(model_def, data="/hdfs/temp")
+couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;""")
+couler.katib.train(model=model_def, hyperparameters={"objective": "multi:softmax", 
+    "eta": 1},  train_data="SELECT * FROM c", validation_data="select * from d")
 ```
 
 ## `couler.katib.train(...)`
 
 Considering Katib itself supports multiple models and frameworks, and more may come in the future, we introduce the following Couler function.
 
 ```python
-def couler.katib.train(model_def=None, hyperparameters={})
+def couler.katib.train(model=None, hyperparameters={}, datasource=None, 
+    train_data=None, validation_data=None)
 ```
+
+The arguments in `couler.katib.train`,
+
+- `model` defines the training model, e.g., `xgboost:gbtree`.
+- `hyperparameters` specifies hyperparameters for model given in `model`.
+- `datasource` specifies the source of training and validation data.
+- `train_data` defines SQL query to fetch training data.
+- `validation_data` defines SQL query to fetch validation data.
+
+## Pipeline
+
+The pipeline from SQL statements to Argo workflow:
+
+- SQLFlow generate `IR` from input SQL statements.
+- `couler_katib_codegen.go` take this `IR` as input and obtains parameters for Katib training job.
```

Comment:
I am not sure if we need a `couler_katib_codegen.go`. What do you think? Can you add more details?  Should we have just a `couler_codegen.go`?  Or, what would be the difference between `couler_katib_codegen.go` and `couler_codegen.go` in your mind? The team would love to listen more about the connection between SQLFlow and Katib.

Reply:
When to fill katib manifest in Argo workflow, we need some information from the input sql statements, for example, tuning hyperparameters. @Yancey1989 suggested having a separate codegen for katib.